jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -162,135 +170,253 @@ data class CodeModernizerSessionContext(
+                        codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
+                        reason = error
+                    )
+                } else {
+                    LOG.info { "$mvnw executed successfully" }
+                    shouldTryMvnCommand = false
+                }
             }
         } catch (e: ProcessNotCreatedException) {
-            val error = "./mvnw failed to execute as its likely not a unix machine"
+            val error = "$mvnw failed to execute as its likely not a unix machine"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                 reason = error
             )
-            LOG.warn { error }
+            LOG.warn(e) { error }
         } catch (e: Exception) {
+            val error = "Unexpected exception when running $mvnw"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
-                reason = e.message
+                reason = error
             )
             when {
                 e.message?.contains("Cannot run program \"./mvnw\"") == true -> {} // noop
                 else -> throw e
             }
+            LOG.error(e) { error }
         }
 
         // 2. maybe execute maven wrapper command
         if (shouldTryMvnCommand) {
-            LOG.warn { "Executing mvn" }
+            LOG.info { "Executing mvn" }
             try {
-                val output = runCommand("mvn")
-                if (output.exitCode != 0) {
-                    LOG.error { "Maven command output:\n$output" }
-                    val error = "The exitCode should be 0 while it was ${output.exitCode}"
+                val installOutput = runInstallCommand("mvn")
+                if (installOutput.exitCode != 0) {
+                    LOG.error { "Maven command output: $installOutput" }
+                    val error = "Maven Install: The exitCode should be 0 while it was ${installOutput.exitCode}"
                     CodetransformTelemetry.mvnBuildFailed(
                         codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                         codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvn,
                         reason = error
                     )
                 } else {
-                    shouldTryMvnCommand = false
-                    LOG.warn { "Maven executed successfully" }
+                    // Copy Dependencies if install is successful
```

Comment:
Sorry I'm just seeing this -- shouldn't we run copy-deps even if install fails? I'm pretty sure I've had a case where copy-deps succeeded after install failed

Reply:
Based on the anecdote of copy dependencies passing after install failing, I'm okay with letting both run at the moment. In the long run, we should fast fail the transformation if maven install itself fails locally, since maven compile is unlikely to pass once uploaded to the backend as well.